web/pdfjs.js

```diff
@@ -21,10 +21,12 @@ if (typeof PDFJSDev !== 'undefined' && PDFJSDev.test('PRODUCTION')) {
   var pdfjsLib;
   // The if below protected by __pdfjsdev_webpack__ check from webpack parsing.
   if (typeof __pdfjsdev_webpack__ === 'undefined') {
-    if (typeof require === 'function') {
-      pdfjsLib = require('../build/pdf.js'); // using a bundler to pull the core
+    if (typeof window !== 'undefined' && window['pdfjs-dist/build/pdf']) {
+      pdfjsLib = window['pdfjs-dist/build/pdf'];
+    } else if (typeof require === 'function') {
+      pdfjsLib = require('../build/pdf.js');
     } else {
-      pdfjsLib = window['pdfjs-dist/build/pdf']; // loaded via html script tag
+      throw 'Neither `require` nor `window` found';
```

Comment:
As [previously mentioned](https://github.com/mozilla/pdf.js/pull/8154#discussion_r105536970), please change this to `throw new Error(...);` instead.
Also, once you've made the changes, please squash the commits such that this is ready for review (see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits for more on this matter).